zulip/zulip/__init__.py

```diff
@@ -221,6 +228,13 @@ def generate_option_group(parser, prefix=''):
 
 def init_from_options(options, client=None):
     # type: (Any, Optional[str]) -> Client
+
+    if options.provision:
+        requirements_path = os.path.abspath(os.path.join(sys.path[0], 'requirements.txt'))
+        if not pip.main(['install', '--upgrade', '--requirement', requirements_path]):
+            print("{color_green}Provisioning successful.{end_color}".format(
+                color_green='\033[92m', end_color='\033[0m'))
```

Comment:
Do you think we want to move the `import pip` into this block, so that we don't add a dependency of the entire library on `pip` being installed?  I guess I'm not entirely sure whether `pip` is ever going to not be installed on a system, but it's not part of Python core.  I guess my idea for how best to do this block would be to add a `try/except` around importing `pip`, with a nice error message if not.  

@eeshangarg FYI.